raft/server.cc

```diff
@@ -692,7 +708,7 @@ future<> server_impl::do_on_leader_with_retries(seastar::abort_source* as, Async
 
     while (true) {
         if (as && as->abort_requested()) {
-            throw request_aborted();
+            throw request_aborted(format("Request aborted while performing action on current leader: {}, previous leader: {}", leader, prev_leader));
```

Comment:
`leader` and `prev_leader` can equal `server_id{}`. In such a case, we log an ID containing only 0s. I don't like such logs because without knowing the code they look like we have a bug. For example, "current leader: 0000..." looks like we expect to have a leader but there is no leader or we have lost it's ID somehow.

Maybe we could log something like `format("Request aborted while performing action on leader, current leader: {}, previous leader: {}", leader ? leader.to_sstring() : "uknown", prev_leader ? prev_leader.to_sstring() : "unknown")`.